src/lib.rs

```diff
@@ -1184,7 +1188,16 @@ pub extern "C" fn zendoo_batch_verify_all_proofs(
     if prioritize { zendoo_pause_low_priority_threads(); }
 
     // Execute batch verification
-    let result = get_batch_verifier_thread_pool(prioritize).install(|| rs_batch_verifier.batch_verify_all(&mut OsRng::default()));
+    let result = match panic::catch_unwind(|| {
```

Comment:
Why a specific catch_unwind here ?

Reply:
Inside this call there is rayon thread pool starts where "unwraps" are used. I haven't found any other correct way how to deal with this unwraps. The situation when thread cannot be spawned is something very specific, so catch_unwind was the only idea how to not crash here and catch some info about raised error

Reply:
I've changed this to more accurate error handling without catch_unwind